keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/ImageService.kt

```diff
@@ -62,4 +62,42 @@ class ImageService(
       .await()
       .lastOrNull()
   }
+
+  suspend fun getNamedImageFromJenkinsInfo(packageName: String, account: String, buildHost: String, buildName: String, buildNumber: String): NamedImage? {
+    return cloudDriverService.namedImages(packageName, account)
+      .await()
+      .lastOrNull { namedImage ->
```

Comment:
I see that this is due to clouddriver's image search endpoint lexically sorting images by name but don't think that guarantees the last element is actually the most recent package. Lexical sorting isn't great for all versioning schemes. I don't think this PR should change, just a note that this could theoretically result in incorrect artifact delivery unless we modify clouddriver to sort by a timestamp if possible.

Reply:
Good note, I'll address this because we have creation date in returned in NamedImage and that will be a really annoying bug to troubleshoot later.